src/components/pages/DetailedProblemView/index.tsx

```diff
@@ -0,0 +1,44 @@
+import { FC, useEffect, useState } from "react";
+import { Link, useParams } from "react-router-dom";
+import { useQuery } from "@apollo/client";
+
+import { GetProblemsQuery, PROBLEMS } from "../ProblemIndex/interface";
+import { ProblemSummaryProps } from "../../molecules/SearchResultTile/interface";
+
+const DetailedProblemView: FC = () => {
+  const params = useParams();
+  const id: any = params.problemId;
+
+  const { data, loading } = useQuery<GetProblemsQuery>(PROBLEMS);
+  const [problems, setProblems] = useState<ProblemSummaryProps[]>([]);
+  const [problemDetails, setProblemDetails] = useState({
+    title: "",
+    author: "",
+  });
+
+  useEffect(() => {
+    if (data?.problems && !loading) {
+      const { problems } = data;
+      const filteredProblems = problems;
+      const currentProblem = problems[id - 1];
+      setProblems(filteredProblems);
+      setProblemDetails({
+        title: currentProblem.title,
+        author: currentProblem.author,
+      });
+    }
+  }, [data?.problems.length, loading]);
+
+  return data?.problems && !loading ? (
```

Comment:
It's used twice, can be extracted to a constant.

Reply:
Do you mean `data?.problems`? Or the whole statement `data?.problems && !loading`?

Reply:
I meant `data?.problems && !loading`, so basically you'd have sth along those lines:
`const problemsLoaded: boolean = data?.problems && !loading;`

Reply:
Done, thanks :)